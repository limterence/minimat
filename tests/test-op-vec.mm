/* test-op-vec: vector operators */
int main() {
  sequence v;
  sequence u;
  sequence w;
  v = 1::5;              /* colon expression */
  u = 100:2:108;
  printseq(u + v);       /* arithmetic */
  printseq(v + u);
  printseq(u - v);
  printseq(v - u);
  printseq(u * v);
  printseq(v * u);
  printseq(u / v);
  printseq(v / u);
  printseq(u % v);
  printseq(-v);
  printseq(v + [1000]);  /* arithmetic with scalar sequence */
  printseq([2000] + v);
  printseq([2000] / v);
}
