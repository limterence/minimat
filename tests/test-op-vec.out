[5 int]
101 104 107 110 113 
[5 int]
101 104 107 110 113 
[5 int]
99 100 101 102 103 
[5 int]
-99 -100 -101 -102 -103 
[5 int]
100 204 312 424 540 
[5 int]
100 204 312 424 540 
[5 int]
100 51 34 26 21 
[5 int]
0 0 0 0 0 
[5 int]
0 0 2 2 3 
[5 int]
-1 -2 -3 -4 -5 
[5 int]
1001 1002 1003 1004 1005 
[5 int]
2001 2002 2003 2004 2005 
[5 int]
2000 1000 666 500 400 
